src/libstd/sys/unix/fs.rs

```diff
@@ -120,21 +123,36 @@ cfg_has_statx! {{
             ) -> c_int
         }
 
-        if !HAS_STATX.load(Ordering::Relaxed) {
-            return None;
-        }
-
-        let mut buf: libc::statx = mem::zeroed();
-        let ret = cvt(statx(fd, path, flags, mask, &mut buf));
-        match ret {
-            Err(err) => match err.raw_os_error() {
-                Some(libc::ENOSYS) => {
-                    HAS_STATX.store(false, Ordering::Relaxed);
-                    return None;
+        match STATX_STATE.load(Ordering::Relaxed) {
+            // For the first time, we try to call on current working directory
+            // to check if it is available.
+            0 => {
+                let mut buf: libc::statx = mem::zeroed();
+                let err = cvt(statx(
+                    libc::AT_FDCWD,
+                    b".\0".as_ptr().cast(),
```

Comment:
Could this perhaps be an empty string and pass `AT_EMPTY_PATH` in the flags?

Reply:
As the comment from @lnicola , maybe we can simply call `statx(0, null(), 0, STATX_ALL, null_mut())` and expecting an `EFAULT` according to [the manual](http://man7.org/linux/man-pages/man2/statx.2.html)?
It costs **only 1.3ns** instead of **580ns** of a successful call.

Reply:
That also seems reasonable to me!

Reply:
That's not exactly what I suggested, but sounds like a good idea anyway :smile:.